### PR TITLE
[P2022-1015] Validate JWT token in Firebase

### PR DESCRIPTION
**Pull request by MaciejJ0**

### CODE REVIEW

File: src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java [hunk 1]
```diff
@@ -51,22 +48,19 @@ protected void doFilterInternal(final HttpServletRequest request, final HttpServ
             if (authorizationHeader != null && authorizationHeader.startsWith("Bearer ")) {
                 try {
                     final String token = authorizationHeader.substring("Bearer ".length());
-                    final DecodedJWT decodedJwt = decodeJwt(token);
 
-                    final String username = decodedJwt.getSubject();
+                    FirebaseToken firebaseToken = firebaseAuth.verifyIdToken(token);
```
Comment (patrykzawadzkiintive) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java:52]: final

**Review by patrykzawadzkiintive — APPROVED** [review]

**Review by edych — APPROVED** [review]

File: src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java [hunk 2]
```diff
@@ -34,6 +30,7 @@
 public class CustomAuthorizationFilter extends OncePerRequestFilter {
 
     private final ObjectMapper objectMapper;
+    private final FirebaseAuth firebaseAuth;
 
     @Value("${retroboard.jwt.secret}")
     private String jwtSecret;
```
Comment (edych) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java:36]: is this variable still in use?

File: src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java (hunk 1 above)
Comment (MaciejJ0) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java]: done

File: src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java (hunk 2 above)
Comment (MaciejJ0) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java:36]: Is not

Comment (MaciejJ0) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/filter/CustomAuthorizationFilter.java]: deleted

**Review by golldfish — APPROVED** [review]

**Pull request merged**